### DISCUSSION TITLE
[Feature Improvements]: Concurrent run of vcpkg with manifest conflict with multithreaded IDE

### DISCUSSION THREAD

**Discussion by Milerius** [microsoft/vcpkg#19131] [Ideas]
****

**Is your feature request related to a problem? Please describe.**

Some IDEs like CLion launch CMake in parallel with different configurations: Debug, Release, RelWithDebugInfo etc.

With the recent addition of manifests this adds a huge user experience problem.

Imagine the following `vcpkg.json` manifest:

```json
{
  "name": "atomicdex-desktop",
  "version-string": "0.3.0",
  "dependencies": [
    "entt",
    "folly",
    "boost-multiprecision",
    "boost-random",
    "boost-lockfree",
    "doctest",
    "fmt",
    "nlohmann-json",
    "range-v3",
    "libsodium",
    "spdlog",
    "date",
    "cpprestsdk",
    "cpp-taskflow"
  ]
}
```

And the fact that i'm using vcpkg as a submodule of my repository

When CLion generates CMake at the same time, two instances of vcpkg install will conflict:

```
Failed to take the filesystem lock on /Users/foo/CLionProjects/atomicDEX-Pro/ci_tools_atomic_dex/vcpkg-repo/.vcpkg-root:
    Resource busy
```

This is a problem because it ruins the user experience, impossible to have a release and debug configuration at the same time in the same workspace

**Proposed solution**

If an instance of vcpkg install is already in progress, wait until the end of it or perform the default behavior (your packages are probably already installed by the first configuration in use)

**Describe alternatives you've considered**

conan has a nice approach to this, with a `--remove_locks` option, can be a good inspiration

**Additional context**

There is my configuration before calling `project()`

```CMake
############################ VCPKG PREREQUISITES ##############
message(STATUS "VCPKG package manager enabled")
set(VCPKG_OVERLAY_PORTS "${CMAKE_CURRENT_SOURCE_DIR}/ci_tools_atomic_dex/vcpkg-custom-ports/ports" CACHE STRING "")
set(_VCPKG_INSTALLED_DIR "${CMAKE_CURRENT_SOURCE_DIR}/ci_tools_atomic_dex/vcpkg-repo/installed")
set(CMAKE_TOOLCHAIN_FILE
        "${CMAKE_CURRENT_SOURCE_DIR}/ci_tools_atomic_dex/vcpkg-repo/scripts/buildsystems/vcpkg.cmake"
        CACHE STRING "")
if (WIN32)
    set(VCPKG_TARGET_TRIPLET "x64-windows" CACHE STRING "")
endif ()
if (APPLE)
    set(VCPKG_APPLOCAL_DEPS OFF CACHE BOOL "")
endif ()
############################ END VCPKG PREREQUISITES ##############################
```

cc @ras0219-msft @Neumann-A

**Comment by BillyONeal** [comment]
@Milerius I believe `--x-wait-for-lock` is the flag you're looking for, though we aren't really sure how that should interact with CMake projects right now.

**Comment by Milerius** [comment]
> @Milerius I believe `--x-wait-for-lock` is the flag you're looking for, though we aren't really sure how that should interact with CMake projects right now.

Yes but this need to be added by the `execute_process` inside `vcpkg.cmake` i guess ? 

https://github.com/microsoft/vcpkg/blob/master/scripts/buildsystems/vcpkg.cmake#L372

**Comment by strega-nil** [comment]
Yeah, probably the cmake build should pass that `--x-wait-for-lock`.
